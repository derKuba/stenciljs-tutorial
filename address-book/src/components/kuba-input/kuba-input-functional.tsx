import { h } from "@stencil/core";

export const KubaInputFunctional = ({
  componentId,
  label,
  type = "text",
  value,
  setter,
}: {
  componentId: string;
  label: string;
  type: "text" | "number";
  value: string;
  setter: Function;
}) => {

  const onInput = (event) => {
    setter({ value: event.target.value });
  };
  return (
    <div class="kuba-input">
      <label htmlFor={componentId}>{label}</label>
      <input type={type} id={componentId} value={value} onInput={onInput} />
    </div>
  );
};
