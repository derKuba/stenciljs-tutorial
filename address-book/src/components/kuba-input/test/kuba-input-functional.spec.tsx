import { h } from "@stencil/core";
import { newSpecPage } from "@stencil/core/testing";
import { KubaInputFunctional } from "../kuba-input-functional";

describe("kuba-input-functional", () => {
  it("renders", async () => {
    const page = await newSpecPage({
      components: [],
      template: () => <KubaInputFunctional
        componentId="street"
        label="Adresse:"
        value={""}
        type={"text"}
        setter={() => { }}
      ></KubaInputFunctional >,
    });
    expect(page.root).toMatchSnapshot();
  });

  it("should trigger onInputEvent", async () => {

    const setterMock = jest.fn();
    const page = await newSpecPage({
      components: [],
      template: () => <KubaInputFunctional
        componentId="street"
        label="Adresse:"
        value={""}
        type={"text"}
        setter={setterMock}
      ></KubaInputFunctional >,
    });

    const event = new Event("input");

    page.root.querySelector("input").dispatchEvent(event);

    expect(setterMock).toHaveBeenCalled();
  });
});
