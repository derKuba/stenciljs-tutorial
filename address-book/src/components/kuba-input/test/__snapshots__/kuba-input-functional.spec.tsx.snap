// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kuba-input-functional renders 1`] = `
<div class="kuba-input">
  <label htmlfor="street">
    Adresse:
  </label>
  <input id="street" type="text" value="">
</div>
`;
